cmake/patches/gsl/1064.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/include/gsl/assert b/include/gsl/assert
+index a6012048..a5c216f8 100644
+--- a/include/gsl/assert
++++ b/include/gsl/assert
+@@ -48,7 +48,7 @@
+ #if defined(__clang__)
+ #define GSL_SUPPRESS(x) [[gsl::suppress("x")]]
+ #else
+-#if defined(_MSC_VER) && !defined(__INTEL_COMPILER)
++#if defined(_MSC_VER) && !defined(__INTEL_COMPILER) && !defined(__NVCC__)
+ #define GSL_SUPPRESS(x) [[gsl::suppress(x)]]
```

Comment:
nvcc doesn't recognize this "gsl" attribute namespace.

Reply:
Hi! I see how onnxruntime is probably the only GSL user to build it with nvcc, but on the other hand the patch is fairly small. Have you considered upstreaming it?